types/spec/axis.d.ts

```diff
@@ -7,94 +7,426 @@ import {
+   */
+  format?: string | SignalRef;
+
+  /**
+   * A title for the axis (none by default).
+   */
+  title?: StringValue;
+
+  /**
+   * The orthogonal offset in pixels by which to displace the axis from its position along the edge of the chart.
+   */
+  offset?: NumberValue;
+
+  /**
+   * The anchor position of the axis in pixels. For x-axes with top or bottom orientation, this sets the axis group x coordinate. For y-axes with left or right orientation, this sets the axis group y coordinate.
+   *
+   * __Default value__: `0`
+   */
+  position?: NumberValue;
+
+  /**
+   * A desired number of ticks, for axes visualizing quantitative scales. The resulting number may be different so that values are "nice" (multiples of `2`, `5`, `10`) and lie within the underlying scale's range.
+   *
+   * For scales of type `"time"` or `"utc"`, the tick count can instead be a time interval specifier. Legal string values are `"millisecond"`, `"second"`, `"minute"`, `"hour"`, `"day"`, `"week"`, `"month"`, and "year". Alternatively, an object-valued interval specifier of the form `{"interval": "month", "step": 3}` includes a desired number of interval steps. Here, ticks are generated for each quarter (Jan, Apr, Jul, Oct) boundary.
+   *
+   * @minimum 0
+   */
+  tickCount?: number | TimeInterval;
+
+  /**
+   * Explicitly set the visible axis tick and label values.
+   */
+  values?: any[] | SignalRef;
+
+  /**
+   * The integer z-index indicating the layering of the legend group relative to other axis, mark, and legend groups.
+   *
+   * @TJS-type integer
+   * @minimum 0
+   */
   zindex?: number;
-  interactive?: boolean;
+
   /**
-   * Boolean value that determines whether the axis should include ticks.
+   * Mark definitions for custom axis encoding.
    */
-  ticks?: boolean;
+  encode?: AxisEncode;
+}
+
+export interface AxisEncode {
   /**
-   * A boolean flag indicating if labels should be included as part of the axis.
+   * Custom encoding for the axis container.
+   */
+  axis?: GuideEncodeEntry<GroupEncodeEntry>;
+  /**
+   * Custom encoding for axis tick rule marks.
+   */
+  ticks?: GuideEncodeEntry<GroupEncodeEntry>;
+  /**
+   * Custom encoding for axis label text marks.
+   */
+  labels?: GuideEncodeEntry<TextEncodeEntry>;
+  /**
+   * Custom encoding for the axis title text mark.
+   */
+  title?: GuideEncodeEntry<TextEncodeEntry>;
+  /**
+   * Custom encoding for axis gridline rule marks.
+   */
+  grid?: GuideEncodeEntry<RuleEncodeEntry>;
+  /**
+   * Custom encoding for the axis domain rule mark.
+   */
+  domain?: GuideEncodeEntry<RuleEncodeEntry>;
+}
+
+export interface BaseAxis<
+  N = NumberValue,
+  NS = number | SignalRef,
+  B = BooleanValue,
+  BNS = number | boolean | SignalRef,
+  S = StringValue,
+  C = ColorValue,
+  FW = FontWeightValue,
+  A = AlignValue,
+  TB = TextBaselineValue,
+  LA = LayoutAlign | SignalRef,
+  SY = SymbolShapeValue,
+  LO = LabelOverlap | SignalRef,
+  DA = number[] | ArrayValueRef
```

Comment:
Why DA?

Reply:
`DashArray`

I don't know whether more descriptive names are better since they can easily be confused with actual types. I think I can add some comments, though. I need to fix some things first, though.